### PR TITLE
feat: move current node account id logging logic into sdk client

### LINKED ISSUE #3296: add exposed node account id into SDKError
### Problem

Follow-up on https://github.com/hashgraph/hedera-json-rpc-relay/pull/3292

### Solution

We should remove the logic added [here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3292/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR732), move it into SDKError, and add the node to the existing error logs.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by natanasow**
**Description**:

Follow-up on https://github.com/hashgraph/hedera-json-rpc-relay/pull/3292

**Related issue(s)**:

Fixes #3296

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/relay/src/lib/eth.ts [hunk 1]
```diff
@@ -1698,6 +1698,11 @@ export class EthImpl implements Eth {
     } catch (e: any) {
       if (e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())) {
         submittedTransactionId = e.transactionId || '';
+
+        // Log the target node account ID, right now, it's populated only for MaxAttemptsOrTimeout error
+        this.logger.info(
+          `${requestDetails.formattedRequestId} Transaction failed to execute against node with id: ${e.nodeAccountId}`
+        );
```
Comment (quiet-node) [inline packages/relay/src/lib/eth.ts:1705]: This means we only log this when the errors are due to a timeout or a dropped connection. I know this is where the discussion originally began, but should we consider sending the error to `sendRawTransactionErrorHandler`? This way, we can log it there and expose the `nodeAccountId` along with all `SdkClient` errors.

File: packages/relay/src/lib/errors/SDKClientError.ts [hunk 2]
```diff
@@ -22,17 +22,19 @@ import { Status } from '@hashgraph/sdk';
...
-  constructor(e: any, message?: string, transactionId?: string) {
+  constructor(e: any, message?: string, transactionId?: string, nodeId?: string | undefined) {
     super(e?.status?._code ? e.message : message);
 
     if (e?.status?._code) {
       this.validNetworkError = true;
       this.status = e.status;
     }
     this.failedTransactionId = transactionId || '';
+    this.nodeAccountId = nodeId;
```
[the remark this answers is not in the document]
Comment (Nana-EC) [reply packages/relay/src/lib/errors/SDKClientError.ts:37]: Agreed a simple one would suffice just as we have in `SDKClientError.spec.ts`

[the remark this answers is not in the document]
Comment (natanasow) [reply packages/relay/src/lib/errors/SDKClientError.ts:37]: Added.

File: packages/relay/src/lib/eth.ts (hunk 1 above)
Comment (natanasow) [reply packages/relay/src/lib/eth.ts]: Done.

**Review by quiet-node — APPROVED** [review]

**Pull request merged**